### DISCUSSION TITLE
[Feature] Mobile App Add Support for Deep Links

### DISCUSSION THREAD

**Discussion by pman07** [immich-app/immich#2881] [Feature Request]
### The feature

Support deep links, something like `immich://<host>/albums/<albumID>`

User clicks on link which opens content within Immich app instead of web browser.

https://developer.android.com/training/app-links/deep-linking

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by tmorgan497** [comment]
I did a lot of searching and finally came across this feature request. I think this feature would be really neat. I use [Gotify](https://gotify.net/) for push notifications from my server, and I'd like to be able to embed a URL into the notification that would take me directly to the specified album or photo.

**Comment by bmachek** [comment]
I think it would be a great feature.
And it's covered by Flutter.

https://docs.flutter.dev/cookbook/navigation/set-up-app-links
https://docs.flutter.dev/cookbook/navigation/set-up-universal-links

**Comment by SteffenAuer** [comment]
Especially considering the E-Mail notifications that are sent out when new assets are added to an album, it would be a great improvement if the "View Album" button in those emails would open the mobile app (if installed) instead of the browser (where the user then likely has to login first,...)

**Comment by daniele-athome** [comment]
I would like to open the Immich app by clicking to a (real) Immich URL (e.g. a photo URL).
However, Android (AFAIK) only supports pre-configured URLs. Maybe an approach similar to the one used by Home Assistant would work? i.e. https://my.home-assistant.io/ I don't know about security concerns though.

**Reply by bo0tzz** [reply]
We already have https://my.immich.app/, and that would play a part in implementing this feature request yes.

**Reply by passabilities** [reply]
> We already have https://my.immich.app/, and that would play a part in implementing this feature request yes.

Is that something that is planned on implementing? Deeplinking that is

**Comment by nathanaelmoyer** [comment]
This would be an awesome feature for creating QR codes to specific shared albums for events. It works fine overall in the browser but it would be great to have people just scan the code and have it open the app directly.

**Comment by bbalz** [comment]
This would be a great feature to have.  Right now the email notifications are nice but since it links to the publically accessible URL, it just opens in a browser and then asks you to log in and view it.  Would love to have it open the native Android app to what the link would be to!

**Comment by chiragpai** [comment]
TLDR; Open in Android app works with a little configuration with Immich server v2.3.1

I just checked current immich Android app's [manifest.xml](https://github.com/immich-app/immich/blob/main/mobile/android/app/src/main/AndroidManifest.xml#L104-L123) and it already has deep links available.

To add "Open in App", customize the email template in your server administration settings, add below HTML template and save.

```Hey {recipientName},
<br/><br/>
You have been given access to <b>{albumName}</b> album by <b>{senderName}</b>
<br/>
<br/>
<table align="center" width="100%" border="0" cellpadding="0" cellspacing="0" role="presentation" style="display:flex!important;justify-content:center!important;margin-top:24px!important;margin-bottom:24px!important"><tbody><tr><td><a href="https://my.immich.app/albums/{albumId}" class="hover_no-underline hover_text-gray-50" style="border-width:1px!important;background-color:#4250AF!important;border-radius:9999px!important;text-decoration:none!important;color:#fffffe!important;font-weight:700!important;text-transform:uppercase!important;line-height:100%;display:inline-block;max-width:100%;mso-padding-alt:0px" target="_blank"><span><!--[if mso]><i style="mso-font-width:0%;mso-text-raise:0" hidden></i><![endif]--></span><span style="max-width:100%;display:inline-block;line-height:120%;mso-padding-alt:0px"><p style="margin-top:12px!important;margin-bottom:12px!important;margin-left:32px!important;margin-right:32px!important;font-size:14px;line-height:24px">Open in app</p></span><span><!--[if mso]><i style="mso-font-width:0%" hidden>&#8203;</i><![endif]--></span></a></td></tr></tbody></table>
```

**Reply by justspacedog** [reply]
It works on iOS devices, but the app must be open beforehand. So for anyone want to create shortcut (for an iPad used as digital photo book/frame): open App, wait 10 seconds, open link (https://my.immich.app/albums/{albumId})

**Comment by cdtl22** [comment]
I’m trying to open a specific photo directly in the iOS app using a deep link, without relying on https://my.immich.app.

Given an asset ID like:

<asset-id>


I can open it via:

https://my.immich.app/photos/<asset-id>


which correctly opens the iOS app using Universal Links.

However, I want to open the photo directly using either:

A custom URL scheme (e.g. immich://...)

Or a self-hosted domain (e.g. https://mydomain/photos/<asset-id>)

Questions:

Does the iOS app support a custom URL scheme for directly opening an asset?

If so, what is the correct deep link format?
Example attempts:

immich://photos/<asset-id>

immich://asset/<asset-id>

Are Universal Links restricted to my.immich.app, or can self-hosted domains be supported?

On Android, the manifest shows pathPrefix="/photos/", so I assume /photos/<asset-id> is the canonical route.

Is there an equivalent supported deep link mechanism on iOS?

**Comment by 1661410107114145152525** [comment]
Does it also work on Android?

**Comment by grassoste** [comment]
I know nothing about Android apps, bit given what @chiragpai showed about the android app manifest, would it be possible to have the url in the manifest being updated with the server/host value? So if my Android app has as server https://mydomain.tld also the manifest uses url? Is it technically doable?

I really like Immich and would be a nice addition! Although it's not such a big issue :)

**Comment by bo0tzz** [comment]
#19232
